akka-persistence-typed/src/main/scala/akka/persistence/typed/state/internal/DurableStateStoreInteractions.scala

```diff
@@ -56,10 +56,26 @@ private[akka] trait DurableStateStoreInteractions[C, S] {
     newRunningState
   }
 
+  protected def internalDelete(ctx: ActorContext[InternalProtocol], cmd: Any): Running.RunningState[S] = {
+
+    val persistenceId = setup.persistenceId.id
+
+    onDeleteInitiated(ctx, cmd)
+
+    ctx.pipeToSelf[Done](setup.durableStateStore.deleteObject(persistenceId)) {
+      case Success(_)     => InternalProtocol.DeleteSuccess
+      case Failure(cause) => InternalProtocol.DeleteFailure(cause)
+    }
+
+    Running.RunningState(revision = 0, state = setup.emptyState, receivedPoisonPill = false)
```

Comment:
We must also think about the query side. A new `DurableStateChange` type must be added as was prepared for in https://github.com/akka/akka/pull/30549/files#diff-43c5d2b2092ca2bc90e24d02701f2494fa9dbb951f396f94cc44738395a8efa1R16

I'm not sure we should reset the revision to 0. Then the next state will be stored with revision 1 and that might be treated as a duplicate, at least that would be problematic in https://github.com/akka/akka-persistence-r2dbc

Maybe we should just continue the the revision numbers even after reset? It would be up to the plugin implementation to keep track of latest used revision number. We already have that requirement for sequence numbers for event sourced.

The way I would implement the deletes in the r2dbc plugin is to not actually delete the row but just clear the payload column. That is similar to how it's done for events. In that way it's easy to continue the sequence numbers even though all events have been "deleted".